docs/GraphRepresentation.md

```diff
@@ -0,0 +1,321 @@
+
+# Graph representation
+
+It is helpful to discuss candidate lists, relevance scores and treatments
+in terms of an __add-on relational graph__ structure,
+as much of the intuition behind the treatments
+is drawn from this representation.
+
+An add-on relational graph is a directed graph $G = (V,E)$ in which:
+
+- each vertex is an add-on appearing in the dataset
+- there is an edge from A to B if add-on B is considered _related_ to add-on A
+- each edge has an associated weight
+    indicating the strength of the relationship.
+
+The data used in generating recommendations can be summarized
+by such a graph, where
+an add-on's candidate list corresponds to its set of neighbours in the graph,
+and relevance scores are given by the edge weights.
+In this setting:
+
+- A relational graph is induced directly by the coinstallation data,
+    considering add-ons related if they are coinstalled,
+    and using coinstallation counts as weights.
+- Treatments can be thought of as transformations
+    which modify add-on relational graphs by
+    adding or deleting edges, or adjusting the edge weights.
+- The final selection of recommendations can itself be viewed as a treatment,
+    which outputs a graph containing, for each add-on A,
+    an edge from A to each of A's N recommendations.
+
+The recommendation algorithm can be rephrased
+in terms of this representation as follows:
+
+1. Load the input data and create the __coinstallation graph__.
+2. Apply each of the specified treatments in turn to the coinstallation graph,
+    resulting in the __treated graph__.
+3. Apply the recommendation selection treatment to the treated graph
+    to produce the __recommendation graph__.
+
+Note that, since the graph is directed, edges do not necessarily run both ways.
+In other words, we allow for a treatment to produce a graph in which
+A is related to B but B is not related to A,
+for example if it removes edges whose weight falls below a threshold.
+However, since the relation of being coinstalled is symmetric,
+the initial coinstallation graph is undirected.
+
+A relational graph has an associated __adjacency matrix__ $C$,
```

Comment:
This is math notation in Latex. It is supposed to get rendered as a nicely formatted math expression, which is usually done using [MathJax](https://www.mathjax.org/).

It's not a standard part of Markdown, and needs to be supported by the Markdown flavour. I don't think it's supported by Github's automatic rendering, but I used it anyway because it's a good way to write precise expressions in text, and it's pretty readable as text. I could instead surround it with backticks or something so that it stands out.